modules/serverless-customize-labels-annotations-routes.adoc

```diff
@@ -0,0 +1,68 @@
+[id="serverless-customize-labels-annotations-routes_{context}"]
+= Customizing labels and annotations for {product-title} routes
+
+{product-title} routes support the use of custom labels and annotations, which can be configured by modifying the `metadata` spec of a Knative service. Custom labels and annotations are propagated from the service to the Knative route, then to the Knative ingress, and finally to the {product-title} route.
+
+The following annotations are supported by Knative Serving:
+
+* Disabling routes, which is specified by the `serving.knative.openshift.io/disableRoute` annotation.
+* Enabling pass-through, which is specified by the `serving.knative.openshift.io/enablePassthrough` annotation.
+
+.Prerequisites
+
+* You must have the {ServerlessOperatorName} and Knative Serving installed on your {product-title} cluster.
+
+.Procedure
+
+. Create a Knative service that contains the label or annotation that you want to propagate to the {product-title} route:
++
+.Example service YAML
+[source,yaml]
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: <service_name>
+  labels:
+    <label_name>: <label_value>
+  annotations:
+    <annotation_name>: <annotation_value>
+...
+----
++
+.Example of disabling routes by using the annotation
+[source,yaml]
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: example-service
+...
+  annotations:
+    "serving.knative.openshift.io/disableRoute": "true"
+...
+----
++
+.Example kn command
+[source,terminal]
+----
+$ kn service create <service_name> \
+  --image=<image> \
+  --annotation <annotation_name>=<annotation_value> \
+  --label <label_value>=<label_value>
+----
+
+. Verify that the {product-title} route has been created with annotation or label that you added, by inspecting the output from the following command:
++
+.Example command for verification
+[source,terminal]
+----
+$ oc get routes.route.openshift.io \
+     -l serving.knative.openshift.io/ingressName=<service_name> \ <1>
+     -l serving.knative.openshift.io/ingressNamespace=<service_namespace> \ <2>
+     -n istio-system -o yaml \
+         | grep -e "<label_name>: \"<label_value>\""  -e "<annotation_name>: <annotation_value>" <3>
+----
```

Comment:
We can't expect to have (grepping) routes if we are using `serving.knative.openshift.io/disableRoute` annotation.

Also, the `istio-system` here should be `knative-serving-ingress`.

Reply:
@nak3 or @markusthoemmes will need to respond to this one - if we update the previous example to passthrough, does that make this step OK?